### PR TITLE
Don't show blank IPs metadata row for containers with no IP

### PR DESCRIPTION
**Pull request by paulbellamy**
Previously we were adding the blank default IP, and showing that (an empty row), but we should skip the row if there are no IPs.

@2opremio ptal

### CODE REVIEW

**Comment by 2opremio** [comment]
When does this happen? If it's due to `-net containerID` (or even `-net host`) we should try to look further instead of simply not showing it.

**Comment by paulbellamy** [comment]
The specific case I was seeing is when `--net=host`. Hmm.. so maybe we should show the host IPs instead, you're saying?

**Comment by 2opremio** [comment]
Yes, or the referenced container IPs if it also happens with `-net
containerId`.

**Comment by paulbellamy** [comment]
@2opremio wdyt?

File: render/topologies.go [hunk 1]
```diff
@@ -116,6 +118,56 @@ var ContainerRenderer = MakeReduce(
...
+func (r containerWithHostIPsRenderer) Render(rpt report.Report) RenderableNodes {
+	containers := r.Renderer.Render(rpt)
+	hosts := MakeMap(
+		MapHostIdentity,
+		SelectHost,
+	).Render(rpt)
+
+	for id, c := range containers {
+		networkMode, ok := c.Node.Latest.Lookup(docker.ContainerNetworkMode)
+		if !ok || networkMode != docker.NetworkModeHost {
```
Comment (2opremio) [inline render/topologies.go]: How about extending this to show the IPs of the parent container when `-net=containerID`?

File: render/topologies.go [hunk 2]
```diff
@@ -116,6 +118,56 @@ var ContainerRenderer = MakeReduce(
...
+			continue
+		}
+
+		ips, ok := c.Node.Sets.Lookup(docker.ContainerIPs)
+		if ok && len(ips) > 0 {
```
Comment (2opremio) [inline render/topologies.go]: Why is this needed? Can the IPs be non-empty when -net=host?

**Comment by 2opremio** [comment]
Also, it would be a good idea to add a test.

File: render/topologies.go (hunk 2 above)
Comment (paulbellamy) [inline render/topologies.go]: a bit belt-and-braces.

File: render/topologies.go (hunk 1 above)
Comment (paulbellamy) [inline render/topologies.go]: Could do. That is a bit more complicated, as you can do `container:weavescope`, `container:a1b2c3d4`, or `container:a1b2c3d4e5f6g7h8i9j0`, so we need to parse apart the network mode string, and lookup the container by name/id/short-id. It already sort-of works (it only shows one of the IPs). I'd say let's open a different issue for that.

Comment (2opremio) [inline render/topologies.go]: Sounds good.

File: render/topologies.go (hunk 2 above)
Comment (2opremio) [inline render/topologies.go]: It makes it a bit confusing when reading (one may think there are actual cases in which this will happen). I would suggest removing it, but I don't have a very strong opinion.

**Comment by 2opremio** [comment]
Minor comment, extra issue and tests-aside LGTM

File: render/topologies.go (hunk 1 above)
Comment (paulbellamy) [inline render/topologies.go]: https://github.com/weaveworks/scope/issues/986

**Comment by paulbellamy** [comment]
Refactored a bit and added test, @2opremio wdyt?

**Comment by 2opremio** [comment]
LGTM, but the linter is failing: _ContainerWithHostIPsRenderer should have comment or be unexported_
